src/toolkits/object_detection/one_shot_object_detection/util/parameter_sampler.cpp

```diff
@@ -64,33 +65,31 @@ void ParameterSampler::set_warped_corners(
 /* Function to sample all the parameters needed to build a transform, and
  * then also build the transform.
  */
-void ParameterSampler::sample(long seed) {
+void ParameterSampler::sample(size_t background_width, size_t background_height,
+                              size_t seed, size_t row_number) {
   double theta_mean, phi_mean, gamma_mean;
-  std::srand(seed);
-  theta_mean = theta_means_[std::rand() % theta_means_.size()];
-  std::srand(seed + 1);
-  phi_mean = phi_means_[std::rand() % phi_means_.size()];
-  std::srand(seed + 2);
-  gamma_mean = gamma_means_[std::rand() % gamma_means_.size()];
+  std::seed_seq seed_seq = {static_cast<int>(seed), static_cast<int>(row_number)};
+  std::mt19937 engine(seed_seq);
+  
+  std::uniform_int_distribution<int> index_distribution(0, INT_MAX);
+  theta_mean = theta_means_[index_distribution(engine) % theta_means_.size()];
+  phi_mean = phi_means_[index_distribution(engine) % phi_means_.size()];
+  gamma_mean = gamma_means_[index_distribution(engine) % gamma_means_.size()];
```

Comment:
It seems strange to take the result modulo the number of means. Why not just generate a random integer in the desired range right off the bat? Could be a useful helper function here: given a non-inclusive upper bound on the range of the random variable, and the random engine, return a random index